packages/cdk-assets/test/private/docker-credentials.test.ts

```diff
@@ -0,0 +1,174 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as mockfs from 'mock-fs';
+import { cdkCredentialsConfig, cdkCredentialsConfigFile, DockerCredentialsConfig, fetchDockerLoginCredentials } from '../../lib/private/docker-credentials';
+import { mockAws, mockedApiFailure, mockedApiResult } from '../mock-aws';
+
+const _ENV = process.env;
+
```

Comment:
Test in here to validate that the docker binary gets called with `--config` when necessary?

Reply:
Sure. That probably belongs in `docker-images.test.ts`. I'll add it there.